File: mad3004-introd-swift/C0711561_F2017_MAD3004_FinalProject/C0711561_F2017_MAD3004_FinalProject/Intern.swift
public class Intern : Employee {
    
    private var schoolName : String
    private var fixedSalary: Double
    
    override init() {
        schoolName = ""
        fixedSalary = 0.0
        super.init()
    }
    
    init(pName: String, pAge: Int, pSchool: String, pFixedSalary: Double) {
        schoolName = pSchool
        fixedSalary = pFixedSalary
        super.init(pName, pAge)
    }
    
    init(pName: String, pAge: Int, pSchool: String, ppV: Vehicle, pFixedSalary: Double) {
        schoolName = pSchool
        fixedSalary = pFixedSalary
        super.init(pName, pAge, ppV)
    }

    override func getTypeOfEmployee() -> String {
        return "Intern"
    }
    
    override func calcEarnings() -> Double {
        return fixedSalary
    }
    
    override func printMyData() -> String {
        return "\(super.printMyData())\n  -School Name: \(schoolName)\n  -Earnings: $\(calcEarnings())"
    }
    
}

File: mad3464-programming-java-i/C0711561_F2017_MAD3464_FinalProject/src/c0711561/f2017/mad3464/finalproject/employee/Intern.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package c0711561.f2017.mad3464.finalproject.employee;

import c0711561.f2017.mad3464.finalproject.Util;
import c0711561.f2017.mad3464.finalproject.vehicle.Vehicle;

/**
 *
 * @author macstudent
 */
public class Intern extends Employee {

    private String schoolName;
    private double fixedSalary;

    public Intern(String name, int age, String schoolName, double fixedSalary) {
        super(name, age);
        this.schoolName = schoolName;
        this.fixedSalary = fixedSalary;
    }
    
    public Intern(String name, int age, String schoolName, double fixedSalary, Vehicle vehicle) {
        super(name, age, vehicle);
        this.schoolName = schoolName;
        this.fixedSalary = fixedSalary;
    }

    @Override
    public String getTypeOfEmployee() {
        return "Intern";
    }

    @Override
    public String printMyData() {
        return super.printMyData()
                + "\n - School Name: " + schoolName
                + "\n - Earnings: " + Util.toCurrencyFormatFrom(calcEarnings());
    }
}
